public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -93,8 +98,27 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
     });
   }
 
+  private _updateDirty = () => {
+    const diff = jsonDiff(vizPanelToPanel(this.state.sourcePanel.resolve()), vizPanelToPanel(this.state.panel));
+    const diffCount = Object.values(diff).reduce((acc, cur) => acc + cur.length, 0);
+    return this.setState({ isDirty: diffCount > 0 });
+  };
+
+  private _setUpChangeSubs() {
+    this._changeSubs = [this.state.panel.subscribeToState(this._updateDirty)];
+
+    if (this.state.panel.state.$data) {
+      this._changeSubs.concat([
+        this.queryRunner.subscribeToState(this._updateDirty),
+        this.dataTransformer.subscribeToState(this._updateDirty),
+      ]);
```

Comment:
I was testing this with live data. I was afraid of causing too many calls to `this._updateDirty` when streaming. But I found the streaming doesn't work when editing a panel in scenes.

Reply:
<img width="1728" alt="Captura de pantalla 2024-05-03 a las 17 12 42" src="https://github.com/grafana/grafana/assets/5699976/a1b331b4-e0da-4126-95aa-274bb6862c5c">

Sorry, I was testing wrong, but yes, this is diffing the panel edit for each of the streaming budgets. Why do we need to check the data state? That data is never persisted, right? Probably we just need to check the transforms. We could use something like we do in DashboardScene:
```js
    this._changeTrackerSub = this._dashboard.subscribeToEvent(
      SceneObjectStateChangedEvent,
      this.onStateChanged.bind(this)
    );
```
Then in the callback we could check the changed stuff, and only run the diff when needed:
```js
  private onStateChanged({ payload }: SceneObjectStateChangedEvent) {
    // If there are no changes in the state, the check is not needed
    if (Object.keys(payload.partialUpdate).length === 0) {
      return;
    }

    // SceneQueryRunner includes the DS configuration
    if (payload.changedObject instanceof SceneQueryRunner) {
      if (!Object.prototype.hasOwnProperty.call(payload.partialUpdate, 'data')) {
        return this.detectSaveModelChanges();
      }
    }
    // SceneDataTransformer includes the transformation configuration
    if (payload.changedObject instanceof SceneDataTransformer) {
      if (!Object.prototype.hasOwnProperty.call(payload.partialUpdate, 'data')) {
        return this.detectSaveModelChanges();
      }
    }
}
```

Reply:
I'll try this